eth/eventdatahandler/event_data_handler.go

```diff
@@ -105,30 +106,30 @@ func (edh *EventDataHandler) HandleBlockEventsStream(blockEventsCh <-chan eventb
 		logger = logger.With(fields.Count(len(tasks)))
 		logger.Info("executing tasks")
 
-		// TODO:
-		// 1) find and remove opposite tasks (start-stop, stop-start, liquidate-reactivate, reactivate-liquidate)
-		// 2) find superseding tasks and remove superseded ones (updateFee-updateFee)
-		for _, task := range tasks {
-			edh.logger.Debug("going to execute task") // TODO: add more task details
-			if err := task(); err != nil {
+		cleanTaskList := cleanTaskList(tasks)
+		for _, task := range cleanTaskList {
+			edh.logger.With(
```

Comment:
`logger` has a block number field, so let's use it instead
```suggestion
			logger = logger.With(
```